tools/serverpod_cli/lib/src/analyzer/models/entity_dependency_resolver.dart

```diff
@@ -30,6 +33,46 @@ class ModelDependencyResolver {
     });
   }
 
+  static void _resolveInheritance(
+    ClassDefinition classDefinition,
+    List<SerializableModelDefinition> modelDefinitions,
+  ) {
+    if (classDefinition.extendsClass != null) {
+      try {
+        var baseClass = modelDefinitions
+            .whereType<ClassDefinition>()
+            .firstWhere((element) =>
+                element.className == classDefinition.extendsClass?.className);
+
+        classDefinition.extendsClass = ExtendsClassInheritanceDefinition(
+          baseClass.className,
+          baseClass.fields,
+          p.joinAll([
+            ...baseClass.subDirParts,
+            '${baseClass.fileName}.dart',
+          ]),
+        );
+      } catch (e) {
+        classDefinition.extendsClass = null;
```

Comment:
Suggestion: If this fails, keep the UnresolvedInheritanceDefinition.

Then we can catch this in the validation and give an error on it.